### PR TITLE
feat: remove username from header

### PR DESCRIPTION
**Pull request by mubbsharanwar**
In the spirit of making registration experience quick and easy (more details on [MVP scope doc](https://2u-internal.atlassian.net/wiki/spaces/Ecomm/pages/653164545/Registration+Login+Progressive+Profiling+Profile+Account+Management+MVP#MVP-Scope-Estimate)), we want to be able to remove username from the registration form. This creates the need to remove it everywhere on the UX too. As part of this initiative, we will be removing username from the headers and making the dropdown consistent with [edx.org](http://edx.org/) site.

**Changes**
Set the flag `HIDE_USERNAME_FROM_HEADER` to hide the username from the header and replace the username with the name for the screen reader text when the flag is enabled.
The flag will be removed when this [PR](https://github.com/openedx/frontend-component-header/pull/465) would be merged

**Screenshots**
<table>
<tr>
<th>Component</th>
<th>Before</th>
<th>After</th>
</tr>
<tr>
<td>Header</td>
<td>
Screenshot 2024-02-13 at 1 31 38 PM
</td>
<td>
Screenshot 2024-02-13 at 1 32 45 PM
</td>
</tr>
<tr>
<td>Studio Header</td>
<td>
Screenshot 2024-02-13 at 1 36 17 PM
</td>
<td>
Screenshot 2024-02-13 at 1 35 17 PM
</td>
</tr>
<tr>
<td>Learning Header</td>
<td>
Screenshot 2024-02-13 at 1 38 22 PM
</td>
<td>
Screenshot 2024-02-13 at 1 39 38 PM
</td>
</tr>
</table>

**Ticket**
[VAN-1804](https://2u-internal.atlassian.net/browse/VAN-1804)

### CODE REVIEW

File: src/learning-header/LearningHeader.test.jsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ describe('Header', () => {
 
   it('displays user button', () => {
     render(<Header />);
-    expect(screen.getByText(authenticatedUser.username)).toBeInTheDocument();
+    expect(screen.getByRole('button', { className: 'd-md-none' })).toBeInTheDocument();
```
Comment (syedsajjadkazmii) [inline src/learning-header/LearningHeader.test.jsx:15]: can't we do `authenticatedUser.name` here?

File: src/studio-header/UserMenu.jsx [hunk 2]
```diff
@@ -8,11 +8,10 @@ import NavDropdownMenu from './NavDropdownMenu';
 import getUserMenuItems from './utils';
 
 const UserMenu = ({
-  username,
+  name,
   studioBaseUrl,
   logoutUrl,
   authenticatedUserAvatar,
-  isMobile,
```
Comment (syedsajjadkazmii) [inline src/studio-header/UserMenu.jsx:15]: why is this removed?

File: src/studio-header/UserMenu.jsx [hunk 3]
```diff
@@ -21,22 +20,21 @@ const UserMenu = ({
...
   ) : (
     <Avatar
       size="sm"
       className="mr-2"
-      alt={username}
+      alt={name}
       data-testid="avatar-icon"
     />
   );
-  const title = isMobile ? avatar : <>{avatar}{username}</>;
```
Comment (syedsajjadkazmii) [inline src/studio-header/UserMenu.jsx:35]: why is this removed?

**Review by syedsajjadkazmii — COMMENTED** [review]
Would this work on the MFEs that are on the openedx organization? like frontend-app-learner-dashboard?

File: src/studio-header/UserMenu.jsx (hunk 3 above)
Comment (mubbsharanwar) [reply src/studio-header/UserMenu.jsx:35]: We don't want to show username for the desktop screen too, that's why this does not make sense.

File: src/studio-header/UserMenu.jsx (hunk 2 above)
Comment (mubbsharanwar) [reply src/studio-header/UserMenu.jsx:15]: we don't need this prop because we are not showing username on any screen(mobile, desktop)

File: src/learning-header/AuthenticatedUserDropdown.jsx [hunk 4]
```diff
@@ -7,14 +7,20 @@ import { faUserCircle } from '@fortawesome/free-solid-svg-icons';
...
 import Notifications from '../Notifications';
 import { selectShowNotificationTray } from '../Notifications/data/selectors';
 import { fetchAppsNotificationCount } from '../Notifications/data/thunks';
 
-const AuthenticatedUserDropdown = ({ enterpriseLearnerPortalLink, intl, username }) => {
+const AuthenticatedUserDropdown = (props) => {
+  const {
+    intl,
+    enterpriseLearnerPortalLink,
+    username,
```
Comment (syedsajjadkazmii) [inline src/learning-header/AuthenticatedUserDropdown.jsx:21]: if username is not being used, can we remove it?

File: src/studio-header/UserMenu.jsx (hunk 2 above)
Comment (syedsajjadkazmii) [reply src/studio-header/UserMenu.jsx:15]: makes sense, just to make sure, its also not being passed to `UserMenu` now?

File: src/learning-header/LearningHeader.test.jsx (hunk 1 above)
Comment (mubbsharanwar) [reply src/learning-header/LearningHeader.test.jsx:15]: `Avatar` is not a DOM element, it's a react component so there is no avatar role. Avatar is eventually using button element.

**Comment by mubbsharanwar** [comment]
> Would this work on the MFEs that are on the openedx organization? like frontend-app-learner-dashboard?

We set the npm alias to indicate from where the header component would be installed. For now, aliase is not added for `frontend-app-learner-dashboard` but we can add it to install frontend-component-header-edx.
Reference to add npm alias :https://github.com/edx/edx-internal/blob/master/frontends/frontend-app-learner-record/prod_config.yml#L6

**Comment by syedsajjadkazmii** [comment]
> > Would this work on the MFEs that are on the openedx organization? like frontend-app-learner-dashboard?
> 
> We set the npm alias to indicate from where the header component would be installed. For now, aliase is not added for `frontend-app-learner-dashboard` but we can add it to install frontend-component-header-edx. Reference to add npm alias :https://github.com/edx/edx-internal/blob/master/frontends/frontend-app-learner-record/prod_config.yml#L6

`frontend-app-learner-dashboard` was just an example, we would need to do this for all MFEs residing in openedX repo.

**Comment by mubbsharanwar** [comment]
> > > Would this work on the MFEs that are on the openedx organization? like frontend-app-learner-dashboard?
> > 
> > 
> > We set the npm alias to indicate from where the header component would be installed. For now, aliase is not added for `frontend-app-learner-dashboard` but we can add it to install frontend-component-header-edx. Reference to add npm alias :https://github.com/edx/edx-internal/blob/master/frontends/frontend-app-learner-record/prod_config.yml#L6
> 
> `frontend-app-learner-dashboard` was just an example, we would need to do this for all MFEs residing in openedX repo.

Yes we can install in openedX repos but we need to add alias in edx-internal, for most of MFEs alias is added, you can check the detail [here](https://2u-internal.atlassian.net/wiki/spaces/ENGAGE/pages/800949354/Remove+username+from+header).
As I already shared the reference with you how to add alias.

File: src/learning-header/AuthenticatedUserDropdown.jsx (hunk 4 above)
Comment (mubbsharanwar) [reply src/learning-header/AuthenticatedUserDropdown.jsx:21]: No, we can't remove username, because it's using in other places like in PROFILE_URL

File: src/studio-header/UserMenu.jsx (hunk 2 above)
Comment (mubbsharanwar) [reply src/studio-header/UserMenu.jsx:15]: This prop is not passing to UserMenu explicitly. For details how the prop are passing check here: src/studio-header/HeaderBody.jsx

**Review by syedsajjadkazmii — APPROVED** [review]
Thankyou

**Review by BilalQamar95 — APPROVED** [review]

**Review by abdullahwaheed — APPROVED** [review]

File: src/Header.jsx [hunk 5]
```diff
@@ -32,6 +32,9 @@ subscribe(APP_CONFIG_INITIALIZED, () => {
     MINIMAL_HEADER: !!process.env.MINIMAL_HEADER,
     ENTERPRISE_LEARNER_PORTAL_HOSTNAME: process.env.ENTERPRISE_LEARNER_PORTAL_HOSTNAME,
     AUTHN_MINIMAL_HEADER: !!process.env.AUTHN_MINIMAL_HEADER,
+    // this flag is introduced to unblock for new release.
+    // the code behind flag would be removed in follow up PR
```
Comment (syedsajjadkazmii) [inline src/Header.jsx:36]: the code would be removed or the flag would be removed?

File: src/learning-header/LearningHeader.test.jsx [hunk 6]
```diff
@@ -15,6 +16,17 @@ describe('Header', () => {
...
 
+  it('displays user button without username', () => {
+    const config = getConfig();
+    mergeConfig({
+      ...config,
+      HIDE_USERNAME_FROM_HEADER: true,
+    });
+    const { queryByTestId } = render(<Header />);
+    const userName = queryByTestId('username');
+    expect(userName).not.toBeInTheDocument();
```
Comment (syedsajjadkazmii) [inline src/learning-header/LearningHeader.test.jsx:27]:
```suggestion
    const username = queryByTestId('username');
    expect(username).not.toBeInTheDocument();
```

File: src/studio-header/StudioHeader.test.jsx [hunk 7]
```diff
@@ -128,6 +130,17 @@ describe('Header', () => {
...
 
+    it('user menu should not contain username', async () => {
+      const config = getConfig();
+      mergeConfig({
+        ...config,
+        HIDE_USERNAME_FROM_HEADER: true,
+      });
+      const { container } = render(<RootWrapper {...props} />);
+      const userMenue = container.querySelector('#user-dropdown-menu');
+      expect(userMenue.textContent).toContain('');
```
Comment (syedsajjadkazmii) [inline src/studio-header/StudioHeader.test.jsx:141]:
```suggestion
      const userMenu = container.querySelector('#user-dropdown-menu');
      expect(userMenu.textContent).toContain('');
```

**Review by syedsajjadkazmii — APPROVED** [review]

File: src/Header.jsx (hunk 5 above)
Comment (mubbsharanwar) [reply src/Header.jsx:36]: flag. comment updated

File: src/DesktopHeader.jsx [hunk 8]
```diff
@@ -69,10 +72,10 @@ class DesktopHeader extends React.Component {
           as={AvatarButton}
           src={avatar}
           alt=""
-          aria-label={intl.formatMessage(messages['header.label.account.menu.for'], { username })}
+          aria-label={intl.formatMessage(messages['header.label.account.menu.for'], { username: usernameOrName })}
           data-hj-suppress
         >
-          {username}
+          {!hideUsername && username}
```
[the remark this answers is not in the document]
Comment (mubbsharanwar) [reply src/DesktopHeader.jsx:78]: No, we don't want to show username or name when HIDE_USERNAME_FROM_HEADER is enabled.

**Review by abdullahwaheed — APPROVED** [review]
